sql/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java

```diff
@@ -110,6 +109,7 @@
 public class TransportShardUpsertAction extends TransportShardAction<ShardUpsertRequest, ShardUpsertRequest.Item> {
 
     private static final String ACTION_NAME = "indices:crate/data/write/upsert";
+    private static final int MAX_RETRY_LIMIT = 100_000; // upper bound to prevent unlimited retries on unexpected states
```

Comment:
This is an arbitrary number, should we make it configurable?

Reply:
yep it's arbitrary but I don't see any need to make it configurable...